apps/els_lsp/src/els_bsp_client.erl

```diff
@@ -61,14 +64,86 @@
 -type pending_request() :: [{request_id(), from()}].
 -type from()            :: {pid(), any()}.
 
+
+%%==============================================================================
```

Comment:
How may of these do we have in the code base? If it's just this one, I'd say it's ok to have it in the module, otherwise it would make sense to factor these into a separate module:
- makes this usable from other code as well
- makes cleanup easier

Reply:
I think it's just the one at the moment - I went ahead and factored it out in the BSP plugin: [rebar3_bsp_gen_server.erl](https://github.com/erlang-ls/rebar3_bsp/blob/d8ed6f020a97ace8f808f7b18eb02224e3e20c08/src/rebar3_bsp_gen_server.erl). It could just be copy/pasted as is. The linked version also has some improvements in the behavior.